lib/Varien/Data/Collection/Db.php

```diff
@@ -586,11 +586,14 @@ public function load($printQuery = false, $logQuery = false)
 
         if (is_array($data)) {
             foreach ($data as $row) {
+                /** @var Mage_Core_Model_Abstract $item */
                 $item = $this->getNewEmptyItem();
                 if ($this->getIdFieldName()) {
                     $item->setIdFieldName($this->getIdFieldName());
                 }
                 $item->addData($row);
+                $item->setOrigData();
+                $item->setLastSavedData();
```

Comment:
I cannot find any reference to the method `setLastSavedData()`. What does it do?